### PR TITLE
Issue-31-spaces-screen-components

### PR DESCRIPTION
**Pull request by BlazingC0der**
spaces page

### CODE REVIEW

File: src/app/(dashboard)/spaces/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+const SpacesLayout = ({ children }: { children: React.ReactNode }) => {
+  return (
+    <div className="flex flex-col space-y-4 w-full">
+      <section className="flex space-x-4 w-full">
+        <Categories />
+      </section>
+      <div className="flex-grow flex space-x-4">
+        <main className="w-3/4 space-y-4 post-feed">{children}</main>
+        <aside className="w-1/4 space-y-4 space-info">
```
Comment (usama-tariq1) [inline src/app/(dashboard)/spaces/layout.tsx]: need to hide the stats below on mobile size and below md 
i would recommend using/creating a hook for this like useBreakpoint
so on mobile size we will be able to actually block component and data loading

File: src/components/dashboard/Spaces/categories.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import { Card } from "../../ui/card"
+import { CardHeader } from "../../ui/card"
+import { CardTitle } from "../../ui/card"
+import { CardContent } from "../../ui/card"
+import { ScrollArea } from "../../ui/scroll-area"
+import { ScrollBar } from "../../ui/scroll-area"
+import { Button } from "../../ui/button"
+import { useState } from "react"
+
+const Categories = () => {
```
Comment (usama-tariq1) [inline src/components/dashboard/Spaces/categories.tsx]: Please follow the issue guide exactly for name components

File: src/components/dashboard/Spaces/community-stats.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Card, CardTitle, CardHeader, CardContent } from "../../ui/card"
+import { Stat } from "./types/spaces"
+
+type CommunityStatsProps = {
+  stats: Stat[]
+}
+
+const CommunityStats: React.FC<CommunityStatsProps> = (props) => {
```
Comment (usama-tariq1) [inline src/components/dashboard/Spaces/community-stats.tsx]: Please follow the issue guide exactly for naming components and file name

File: src/components/dashboard/Spaces/trending-topics.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import { CardContent } from "../../ui/card"
+import { Badge } from "../../ui/badge"
+import { TrendingUp } from "lucide-react"
+import { Topic } from "./types/spaces"
+
+type TendingTopicProps = {
+  topics: Topic[]
+}
+
+const TrendingTopics: React.FC<TendingTopicProps> = (props) => {
```
Comment (usama-tariq1) [inline src/components/dashboard/Spaces/trending-topics.tsx]: Please follow the issue guide exactly for naming components and file name

File: src/components/dashboard/Spaces/upcoming-events.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { Badge } from "../../ui/badge"
+import { Calendar } from "../../ui/calendar"
+import { Card, CardHeader, CardTitle, CardContent } from "../../ui/card"
+import { Event } from "./types/spaces.d"
+
+type UpcomingEventsProps = {
+  events: Event[]
+}
+
+const UpcomingEvents: React.FC<UpcomingEventsProps> = (props) => {
```
Comment (usama-tariq1) [inline src/components/dashboard/Spaces/upcoming-events.tsx]: Please follow the issue guide exactly for naming components and file name

**Review by usama-tariq1 — CHANGES_REQUESTED** [review]
Please review requested changes. let me know if you like to have a separate issue for this. 
dont forget to pull you branch before working as i just synced it with development

File: src/app/(dashboard)/spaces/layout.tsx (hunk 1 above)
Comment (BlazingC0der) [reply src/app/(dashboard)/spaces/layout.tsx]: I've changed the useIsMobile hook to accomodate custom breakpoints

**Comment by usama-tariq1** [comment]
Feels very unfinished 
image
image

**Comment by usama-tariq1** [comment]
need some hard look on responsiveness it is the good time to do it now :)
image
image

**Comment by usama-tariq1** [comment]
@BlazingC0der  please start over with this screen , doesn't feel good 
image

**Comment by usama-tariq1** [comment]
@BlazingC0der  please make sure to pull it fresh out of development and sync everytime you submit a pull request with development or base branch targeted

**Pull request closed without merging**